### PR TITLE
30 Document Project Structure 

### PR DESCRIPTION
**Pull request by owitiA**
#### Pre-Submission Checklist

- [x] Your pull request targets the `stage` branch of Kudobuzz.
- [x] Branch name is descriptive. 
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Checklist:

- [ ] All test are passing. 
- [ ] Tested changes locally.
- [x] Api Documentation follows api standards.
- [x] Improved documentation or readme where necessary. 

#### What does this pr do?
This  shows the layout of the project structure to be followed and briefly explains what should be contained in every segment of the project

### CODE REVIEW

File: docs/FOLDERS.md [hunk 1]
```diff
@@ -24,37 +24,47 @@
 Contains:
  1. Actions, scripts that interact with the services in [services](#services) folder.
  2. Endpoints, exposing the service to other services.
+ 3. Handles the validations and assertions of the requests
 
 ### api/common
```
Comment (obengwilliam) [inline docs/FOLDERS.md]: in the common u have have something like middlewares too

Comment (obengwilliam) [reply docs/FOLDERS.md]: the middlewares are express middlewares that can use used by any sub router

File: docs/FOLDERS.md [hunk 2]
```diff
@@ -24,37 +24,47 @@
...
 This folder contains all the logic that is shared across multiple files in the parent folder ([api](#api)).
 
 ## config
 This is where configurations are loaded, configurations could be loaded from the environment variables either in production or in development based on where the service has been spawned.
```
Comment (obengwilliam) [inline docs/FOLDERS.md]: Configurations can loaded in any environment. But you should basically emphasize two things
1. how configs are grouped into components
2. and the fact that they basically mainly contains configs for our various process types

File: docs/FOLDERS.md [hunk 3]
```diff
@@ -24,37 +24,47 @@
...
+i.e.
+The process.env files is in the config files 
+The configenables you to test when an app starts to test if all the variables are available if not it throws an error hence one can never deploy an app with missing variables.
+
```
Comment (obengwilliam) [inline docs/FOLDERS.md]: here u can mention api.md too

File: docs/FOLDERS.md [hunk 4]
```diff
@@ -24,37 +24,47 @@
...
 
 ## docs
 All the docs for this service should be here.
 
 ## events
 This is where all the events handlers should be located. Events are very ideal for queue related operations.
+An event triggers an action
```
Comment (obengwilliam) [inline docs/FOLDERS.md]: i dont think this is necessry

File: docs/FOLDERS.md [hunk 5]
```diff
@@ -24,37 +24,47 @@
...
 
 ## docs
 All the docs for this service should be here.
 
 ## events
 This is where all the events handlers should be located. Events are very ideal for queue related operations.
+An event triggers an action 
 
 ## lib
 Contains the wrappers to external libraries.
```
Comment (obengwilliam) [inline docs/FOLDERS.md]: and utilities

File: docs/FOLDERS.md [hunk 6]
```diff
@@ -24,37 +24,47 @@
...
 
+
 ## scripts
 This contains all scripts that are run once, E.g. Migration scripts, indice creators, administrative scripts, local test utitlites or any build configs.
 
 ## services
-This is where all the business code should be located.
+This is where all the business code should be located.It contains the business logic and application specific logic as well
```
Comment (obengwilliam) [inline docs/FOLDERS.md]: it seems it has repeated, i am not sure application specific logic means i think it probably also means business logic right

File: docs/FOLDERS.md [hunk 7]
```diff
@@ -24,37 +24,47 @@
...
+for example encrypting and decrypting a token as well as communication with the models
 
 ## tests
 This contains test configurations.
```
Comment (obengwilliam) [inline docs/FOLDERS.md]: like ? you can mention the sub folders
like test/setup.js
       test/utils.js
       test/fixtures/

File: docs/FOLDERS.md [hunk 8]
```diff
@@ -24,37 +24,47 @@
...
 
 ## services
-This is where all the business code should be located.
+This is where all the business code should be located.It contains the business logic and application specific logic as well
+for example encrypting and decrypting a token as well as communication with the models
 
 ## tests
 This contains test configurations.  
 
 ### services/models
```
Comment (obengwilliam) [inline docs/FOLDERS.md]: just like u changed up there we now have models in the upper layer

File: docs/FOLDERS.md (hunk 1 above)
Comment (owitiA) [reply docs/FOLDERS.md]: added that

File: docs/FOLDERS.md (hunk 6 above)
Comment (owitiA) [reply docs/FOLDERS.md]: yes

File: docs/FOLDERS.md (hunk 2 above)
Comment (owitiA) [reply docs/FOLDERS.md]: added

**Comment by owitiA** [comment]
kindly add express-api-starter to codeship

**Comment by obengwilliam** [comment]
>kindly add express-api-starter to codeship

@owitiA if there is an ops issue create an issue on the ops

**Comment by owitiA** [comment]
> > kindly add express-api-starter to codeship
> 
> @owitiA if there is an ops issue create an issue on the ops

added

**Comment by obengwilliam** [comment]
@owitiA please ready up on the changes

**Comment by obengwilliam** [comment]
@owitiA  when you add an issue on the ops please ref it here

**Pull request merged**